### ISSUE TITLE
JPG thumbnails not being generated when serving website on a non-FQDN

### ISSUE THREAD

**Issue by maniqui** [jazzband/sorl-thumbnail#339]
This issue took a away a few hours of my life, so it's my duty to report it. That said, I'm still not sure what's exactly going on here (I'm more a front-end dev than a back-end one).

So far, the project is pretty simple: a single web page, all content is hardcoded directly in the template, database is only used by sorl-thumbnail to store the key-values. Images are provided to the sorl-thumbnail tag as if they were external images (basically, I concatenate some strings to get the full URL for an image stored somewhere in the `/static/` folder).
Like this:

```
{% with 'http://'|add:request.get_host|add:STATIC_URL|add:'website/images/someproduct.jpg' as source %}
    <img  class="ProductCard-img"  src="{% thumbnail source '450' as im %}{{ im.url }}{% endthumbnail %}" />
{% endwith %}
```

I'm developing locally, at my home, on a desktop/server running Debian Testing.
First, let me say that when running the project using the Django development server on a virtualenv, sorl-thumbnail works flawlessly. It doesn't matter if the website is served at `http://localhost` or `http://project.localhost` or `http://project.localhost.lan` or even at `http://project.myhomeserver.com`. It just works.

Problem arises when I try to "deploy" it. By "deploy", I just mean to create a copy of the website, running on the _same_ machine, but this copy now running as if it was a "production" copy. This means that I've set up a vhost in Apache (following the official documentation), and set `DEBUG` to `False`.

So, my first try was to serve the website at `http://project-vhost.localhost`. Page loaded successfully (including static assets, like CSS and JS), but most sorl-thumbnail-related images were missing. Why "most"? This was the exception: any `sorl-thumbnail`'ed image in PNG format was processed (its thumbnails generated and cached in `/media/cache`) and served by Apache, as expected. But all JPG files were missing, the browser showing the classic frame of an image not loaded. When right-clicking the image and opening it on a new tab, the URL was one corresponding to a (sorl-thumbnail generated) thumbnail, like `http://project-vhost.localhost/media/cache/a7/1d/a71d9e08969c87f570aaddb4e03ce52b.jpg`, and Apache returned a 404 Not Found.
Of course, I checked if the corresponding file existed at `/path/to/project/media/cache/a7/1d/a71d9e08969c87f570aaddb4e03ce52b.jpg`, and no, it wasn't there.
My first guess was: "if PNG files are processed and JPG files aren't, then this issue is related with some OS image library and/or Pillow". So I reinstalled the OS libraries and reinstalled Pillow, but the problem remained. Then I tried installing [other image librares](http://sorl-thumbnail.readthedocs.org/en/latest/requirements.html#image-library), but no luck.
By that time, I was really tired of tinkering, and, also, already run out of ideas

The next morning, while connected to the Internet from outside of my home, I decided to do some remote work on my home server, to see if I could finally figure out what was going on. 
So I opened the browser and visited the website using its public URL: `http://project-vhost.myhomeserver.com`.
And you know what? **All images loaded flawlessly**.
So, after the WTF moment, and when I returned home, I tried again.
These are the results:
- Visiting `http://project-vhost.localhost` -> JPG images not generated and not served
- Visiting `http://project-vhost.myhomeserver.com` -> JPG images generated, everything working as expected.
- Visiting `http://project-vhost.localhost.lan` -> Idem. JPG images generated, everything working as expected.

So, bottom line, my guess is: there seems to be an issue between sorl-thumbnail, images library and non-FQDN when doing a proper deploy of Django behind Apache. No issues when on Django development server.

Can anyone try to reproduce this?
Feel free to ask me for more info.
